commit b50e8c5698be34c39b3ab4cdf4b4daef6d8a87e3
Author: q66 <q66@chimera-linux.org>
Date:   Thu Jun 1 03:13:52 2023 +0200

    use prebuilt manpages
    
    refreshed for 2.41 (block moved; manadocs are File objects, so
    fs.replace_suffix instead of str.split; keep the asciidoctor
    find_program so po-man still sees a defined variable)

diff --git a/meson.build b/meson.build
index 1537f5e..1c4e76e 100644
--- a/meson.build
+++ b/meson.build
@@ -4040,33 +4040,12 @@ if build_libuuid
   }
 endif
 
+# asciidoctor is intentionally absent from the bldroot: the dist tarball
+# ships prebuilt pages, and po-man's asciidoctor.found() test stays false.
 asciidoctor = find_program('asciidoctor', required : false)
-if asciidoctor.found()
+if true
   foreach adoc : manadocs
-    name = fs.name(adoc)
-    page = fs.stem(adoc)
-    section = page.split('.')[-1]
-    mandirn = mandir / 'man' + section
-    input = adoc
-
-    custom_target(
-      page,
-      command : [ asciidoctor,
-	  '-b', 'manpage',
-	  '-a', 'VERSION=' + meson.project_version(),
-	  '-a', 'release-version=' + meson.project_version(),
-	  '-a', 'ADJTIME_PATH=/etc/adjtime',
-	  '-a', 'package-docdir=' + docdir,
-	  '--base-dir=' + meson.current_source_dir(),
-          '--destination-dir=' + meson.current_build_dir(),
-          '--load-path', '@SOURCE_ROOT@/tools',
-          '--require', 'asciidoctor-includetracker',
-	  '@INPUT@'],
-      input : input,
-      output : [page],
-      depfile : page + '.deps',
-      install: true,
-      install_dir : mandirn)
+    install_man(fs.replace_suffix(adoc, ''))
   endforeach
 
   foreach link_name, target : manlinks
